core/src/banking_stage/transaction_scheduler/scheduler_controller.rs

```diff
@@ -105,8 +105,8 @@ impl<C: LikeClusterInfo, R: ReceiveAndBuffer> SchedulerController<C, R> {
             self.timing_metrics
                 .maybe_report_and_reset_slot(new_leader_slot);
 
-            self.process_transactions(&decision)?;
             self.receive_completed()?;
+            self.process_transactions(&decision)?;
```

Comment:
Behavioral change: receive completed transactions just before scheduling instead of just after.